src/UserRepo.js

```diff
@@ -25,12 +26,21 @@ class UserRepo extends Repo  {
       return dataPoint.id === id;
     });
   }
-  
+
+/* BECAUSE OF THE findCurrentUser() METHOD, THIS HAD TO BE REFACTORED
+SINCE THE RANDOM NUMBER WAS SOMETIMES THE SAME AS THE INDEX AND SOMETIMES NOT.
+THIS WAS RESULTING IN THE TEST PASSING SOMETIMES AND FAILING OTHER TIMES.*/
   getRandomNumber() {
-    return Math.floor(Math.random() * 50)
+    let randomNum = Math.floor(Math.random() * 2)
+    if (randomNum === 0) {
+      return 3000
```

Comment:
Sorry if this is part of what's thus far unfinished and if the comment is undo.
Are you picking a random number between 0 and 2 and then returning 3000 if it's 0? I don't quite follow.
Depending on the first comment we may just be able to nix this.